### PR TITLE
Rename CorrelationContext to Baggage

### LINKED ISSUE #536: Rename CorrelationContext to Baggage (again)
The W3C WG is leaning towards changing the `Correlation-Context` header name to `baggage` (https://github.com/w3c/correlation-context/issues/17).

What does this mean for OTel?

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

Fixes #536

### CODE REVIEW

File: specification/baggage/api.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+`Baggage` MAY be propagated across process boundaries or across any arbitrary boundaries
+(process, $OTHER_BOUNDARY1, $OTHER_BOUNDARY2, etc) for various reasons.
+
+### Serialization
+
+Until the [W3C Baggage](https://w3c.github.io/baggage/) specification is recommended for use, OpenTelemetry `Baggage` implementations MUST be serialized according to the [editor's draft of W3C Correlation Context as of March 27, 2020](https://github.com/w3c/correlation-context/blob/c974664b9ab4d33af6355f1f7f03a2d52c89a99e/correlation_context/HTTP_HEADER_FORMAT.md) using a vendor-specific header name to avoid collisions with the W3C Baggage specification should it change in the future.
+
+#### Header Name
+
+`Baggage` implementations MUST use the header name `otcorrelations`.
```
[the remark this answers is not in the document]
Comment (SergeyKanzhelev) [reply specification/baggage/api.md:110]: let's track it separately. I think we should keep the renaming PR small. @cijothomas can you file this issue?

**Review by carlosalberto — APPROVED** [review]

**Review by ktoso — APPROVED** [review]
Very happy to see `Baggage` name be adopted in otel; it's also what we set out with and arrived at in (server side) Swift, great that all terms are finally getting fleshed out 👍  

(`BaggageContext` could also be good (in fact, that's what we do today), but `Baggage` is good, especially since it is carried in a context and one would want to avoid people writing "context.context" which we've seen some, so taking the context part of the name away helps in clarity).

👍

**Comment by SergeyKanzhelev** [comment]
this has enough approvals. Letting this PR cool off for another day and will merge tonight if no objections.

**Comment by carlosalberto** [comment]
@SergeyKanzhelev CHANGELOG entry is missing (but you that can be done in a follow up as well, in order to merge this change fast).

**Comment by SergeyKanzhelev** [comment]
@bogdandrutu can you update the changelog?

File: specification/baggage/api.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (SergeyKanzhelev) [reply specification/baggage/api.md:110]: @cijothomas have you created an issue?

[the remark this answers is not in the document]
Comment (SergeyKanzhelev) [reply specification/baggage/api.md:110]: https://github.com/open-telemetry/opentelemetry-specification/issues/879

[the remark this answers is not in the document]
Comment (cijothomas) [reply specification/baggage/api.md:110]: sorry i didn't hit submit:(
Thanks for creating the issue!

**Pull request merged**